File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    clothdict = {}
    for i in clothes :
        if i[1] in clothdict.keys() : 
            clothdict[i[1]] += 1
        else :
            clothdict[i[1]] =1
    
    for i in clothdict : 
        answer = answer * (clothdict[i]+1)

    answer = answer -1 
    
    return answer

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public Map<String, Set<String>> map = new HashMap<>();
    public int solution(String[][] clothes) {
        int answer = 1;
        
        for (int i = 0 ; i<clothes.length; i++)
        {
            String [] now  = clothes[i];
            String val = now[0];
            String key = now[1];
        
            if(!map.keySet().contains(key))
            {
                Set<String>newset = new HashSet<>();
                newset.add(val);
                map.put(key,newset);
            }else
            {
                map.get(key).add(val);
            }
        }
        
        for (String key : map.keySet())
        {
            answer *= (map.get(key).size()+1);
        }
        
        return answer-1;
    }
}